flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/streaming/util/TestStreamEnvironment.java

```diff
@@ -124,33 +124,33 @@ private static void randomizeConfiguration(MiniCluster miniCluster, Configuratio
         }
 
         // randomize ITTests for enabling state change log
-        if (STATE_CHANGE_LOG_CONFIG.equalsIgnoreCase(STATE_CHANGE_LOG_CONFIG_ON)) {
-            if (!conf.contains(StateChangelogOptions.ENABLE_STATE_CHANGE_LOG)) {
+        if (!conf.contains(StateChangelogOptions.ENABLE_STATE_CHANGE_LOG)) {
+            if (STATE_CHANGE_LOG_CONFIG.equalsIgnoreCase(STATE_CHANGE_LOG_CONFIG_ON)) {
                 conf.set(StateChangelogOptions.ENABLE_STATE_CHANGE_LOG, true);
-                miniCluster.overrideRestoreModeForChangelogStateBackend();
-            }
-        } else if (STATE_CHANGE_LOG_CONFIG.equalsIgnoreCase(STATE_CHANGE_LOG_CONFIG_RAND)) {
-            boolean enabled =
-                    randomize(conf, StateChangelogOptions.ENABLE_STATE_CHANGE_LOG, true, false);
-            if (enabled) {
-                // More situations about enabling periodic materialization should be tested
-                randomize(
-                        conf,
-                        StateChangelogOptions.PERIODIC_MATERIALIZATION_ENABLED,
-                        true,
-                        true,
-                        true,
-                        false);
-                randomize(
-                        conf,
-                        StateChangelogOptions.PERIODIC_MATERIALIZATION_INTERVAL,
-                        Duration.ofMillis(100),
-                        Duration.ofMillis(500),
-                        Duration.ofSeconds(1),
-                        Duration.ofSeconds(5));
-                miniCluster.overrideRestoreModeForChangelogStateBackend();
+            } else if (STATE_CHANGE_LOG_CONFIG.equalsIgnoreCase(STATE_CHANGE_LOG_CONFIG_RAND)) {
+                randomize(conf, StateChangelogOptions.ENABLE_STATE_CHANGE_LOG, true, false);
             }
         }
+
+        // randomize periodic materialization when enabling state change log
+        if (conf.get(StateChangelogOptions.ENABLE_STATE_CHANGE_LOG)) {
```

Comment:
If the `PERIODIC_MATERIALIZATION_ENABLED` is already set in the test, would this override the original value?

Reply:
Thanks for your suggestion.
I have also applied to it.